pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -77,6 +77,132 @@ func createEcNegateHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
+// to the final result to ensure safety and prevent division by zero errors.
+//
+// `DivModeNSafeDivPlusOne` assigns the result as `value` in the current scope.
+func newDivModNSafeDivPlusOneHint() hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DivModNSafeDivPlusOne",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> value = k_plus_one = safe_div(res * b - a, N) + 1
+			valueBig := new(big.Int)
+
+			resBig, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "res")
+			if err != nil {
+				return err
+			}
+
+			aBig, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "a")
+			if err != nil {
+				return err
+			}
+
+			bBig, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "b")
+			if err != nil {
+				return err
+			}
+
+			nBig, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "N")
+			if err != nil {
+				return err
+			}
+
+			valueBig.Mul(resBig, bBig)
+			valueBig.Sub(valueBig, aBig)
+
+			newValueBig, err := secp_utils.SafeDiv(valueBig, nBig)
+			if err != nil {
+				return err
+			}
+
+			newValueBig.Add(&newValueBig, big.NewInt(1))
+
+			return ctx.ScopeManager.AssignVariable("value", &newValueBig)
+		},
+	}
+}
+
+func createDivModNSafeDivPlusOneHinter() (hinter.Hinter, error) {
+	return newDivModNSafeDivPlusOneHint(), nil
+}
+
+// DivModNPackedDivModExternalN computes the div_mod operation for a given packed value.
+// `newDivModNPackedDivModExternalN` takes 2 operander as arguments
+//   - `a` is the value that will be packed and taken prime
+//   - `b` is the value that will be packed and taken prime
+//
+// `DivModNPackedDivModExternalN` assigns the result as `value` in the current scope.
+func newDivModNPackedDivModExternalN(a, b hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DivModNPackedDivModExternalN",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import pack
+			//> from starkware.python.math_utils import div_mod, safe_div
+			//> a = pack(ids.a, PRIME)
+			//> b = pack(ids.b, PRIME)
+			//> value = res = div_mod(a, b, N)
+
+			aAddr, err := a.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			aValues, err := vm.Memory.ResolveAsBigInt3(aAddr)
+			if err != nil {
+				return err
+			}
+
+			aBig, err := secp_utils.SecPPacked(aValues)
+			if err != nil {
+				return err
+			}
+
+			bAddr, err := b.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			bValues, err := vm.Memory.ResolveAsBigInt3(bAddr)
+			if err != nil {
+				return err
+			}
+
+			bBig, err := secp_utils.SecPPacked(bValues)
+			if err != nil {
+				return err
+			}
+
+			nBig, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "N")
+			if err != nil {
+				return err
+			}
+
+			newValueBig, err := secp_utils.Divmod(&aBig, &bBig, nBig)
+			if err != nil {
+				return err
+			}
+
+			return ctx.ScopeManager.AssignVariable("value", &newValueBig)
```

Comment:
add `res` to the scope as well?

Reply:
I think the next hint you implemented uses that to calculate the `value`

Reply:
where is it needed to add `res` in scope?

`let (res) = nondet_bigint3();` uses `value` in scope to create `res` variable outside the hint

Reply:
I added `res` to the scope and also checked its value in the unit test, if it is not required I can revert those changes

Reply:
@TAdev0 the first hint assigns value to the `res` in scope, so that `value = k_plus_one = safe_div(res * b - a, N) + 1` this one can use that value

Reply:
were integration tests passing before adding `res` in scope @danielcdz ? if yes its weird

Reply:
Yes they were passing @TAdev0

Reply:
@TAdev0 @MaksymMalicki I noticed that when I added res to the scope the integration test failed with this error: 
![image](https://github.com/user-attachments/assets/ac0ef48a-e209-4944-b2f7-41e045df6522)
when not it passes normally

Reply:
ok i don't know whats wrong for now, i'm looking into it.

What i can tell is that integration test uses `DivModNPackedDivmodV1` hint already implemented, which is:

```
			//> from starkware.cairo.common.cairo_secp.secp_utils import N, pack
			//> from starkware.python.math_utils import div_mod, safe_div
			//>
			//> a = pack(ids.a, PRIME)
			//> b = pack(ids.b, PRIME)
			//> value = res = div_mod(a, b, N)
```

in this one, we add both `res`  and `value` in scope

`DivModSafeDiv` without the +1 is not used in the tests

Reply:
@MaksymMalicki  it worked without adding `res` because the test file uses as first hint a version we already implemented that adds `res` in scope. And because its the same value and because scope is persistent, we had it.